packages/flutter/lib/src/widgets/platform_view.dart

```diff
@@ -581,6 +581,129 @@ class _UiKitPlatformView extends LeafRenderObjectWidget {
   }
 }
 
+/// The parameters used to create a [PlatformViewController].
+///
+/// See also [CreatePlatformView] which uses this object to create a [PlatformViewController].
+class PlatformViewCreationParams {
+
+  const PlatformViewCreationParams._({
+    @required this.id,
+    @required this.onPlatformViewCreated}):assert(id != null),
+                          assert(onPlatformViewCreated != null);
+
+  /// The auto generated id for the newly created platform view.
+  ///
+  /// [PlatformViewController.viewId] should match this id.
+  final int id;
+
+  /// Notifies when the PlatformView is ready.
+  ///
+  /// Must be invoked as soon as the embedded platform view is ready.
+  /// For example, on iOS, this has to be called when we know the `UIView` is initialized.
+  final PlatformViewCreatedCallback onPlatformViewCreated;
+}
+
+/// The factory to construct a [PlatformViewSurface].
+///
+/// It is required when constructing a [PlatformViewLink].
+typedef PlatformViewSurfaceFactory = PlatformViewSurface Function(BuildContext context, PlatformViewController controller);
+
+/// Constructs a [PlatformViewController].
+///
+/// The implementer of a new platform view is responsible to implement this method when constructing a [PlatformViewLink].
```

Comment:
Constructs a [PlatformViewController].

The [PlatformViewController.id] field of the created controller must match the value of the `params` [PlatformViewCreationParams.id]  field.

See also [PlatformViewLink.onCreate].